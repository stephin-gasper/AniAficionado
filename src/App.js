import React, {useCallback} from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {ThemeProvider} from 'styled-components/native';

import Home from 'screens/home';
import theme from 'constants/theme';
import Header from './components/header';

const Stack = createNativeStackNavigator();

const App = () => {
  const MemoizedHeader = useCallback(() => <Header />, []);
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>
        <Stack.Navigator
          initialRouteName="Home"
          screenOptions={{
            header: MemoizedHeader,
          }}>
          <Stack.Screen name="Home" component={Home} />
        </Stack.Navigator>
      </NavigationContainer>
    </ThemeProvider>
  );
};

export default App;
