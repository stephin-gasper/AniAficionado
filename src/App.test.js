import React from 'react';
import {View as MockView} from 'react-native';
import {render} from '@testing-library/react-native';

import App from './App';

jest.mock('screens/home', () => () => <MockView testID="homeContainer" />);

describe('tests for App', () => {
  it('should render component', () => {
    expect(render(<App />)).toMatchSnapshot();
  });

  it('should render home screen', () => {
    const {getByTestId} = render(<App />);

    expect(getByTestId('homeContainer')).toBeDefined();
  });

  it('should render header', () => {
    const {getByTestId} = render(<App />);

    expect(getByTestId('header')).toBeDefined();
  });
});
