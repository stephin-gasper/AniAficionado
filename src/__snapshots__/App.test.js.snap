// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for App should render component 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
      >
        <View
          testID="header"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          testID="homeContainer"
        />
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
